sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -233,6 +238,15 @@ case class StateStoreMetrics(
     memoryUsedBytes: Long,
     customMetrics: Map[StateStoreCustomMetric, Long])
 
+case class StateStoreCheckpointInfo(
+    partitionId: Int,
+    batchVersion: Long,
+    // The checkpoint ID for a checkpoint at `batchVersion`. This is used to identify the checkpoint
+    checkpointId: Option[String],
```

Comment:
If we use a String, we need to mention that it's not necessarily _one_ checkpoint ID. It could be many, comma-separated.

But to be honest, I don't think we should be using `String` here, because it's ambiguous. Is it 1 checkpoint? 4 checkpoints? You cannot simply tell by looking at the code. The naming is also off in the case of multiple checkpoints; it's `StateStore*s*CheckpointInfo`.

I think it makes more sense for us to return, all the way through the accumulator, a `Seq[String]`. Then, the only place that the merging should happen is inside of `def getCheckpointInfo` inside of `StateStoreWriter`. This avoids us from awkwardly having one-off merging logic inside of the s/s join, even though I know it's the only place.